nca/Parsers/IstioSidecarYamlParser.py

```diff
@@ -154,12 +163,13 @@ def _check_and_save_sidecar_if_top_priority(self, curr_sidecar):
             self.namespace.prior_default_sidecar = curr_sidecar
             return
 
-        for peer in curr_sidecar.selected_peers:
-            if peer.prior_sidecar:  # this sidecar is not first one
-                self.warning(f'Peer "{peer.full_name()}" already has a Sidecar configuration selecting it. Sidecar: '
-                             f'"{curr_sidecar.full_name()}" will not be considered as connections for this workload')
-                continue
-            peer.prior_sidecar = curr_sidecar
+        prior_referenced_by_label = curr_sidecar.selected_peers & self.peers_referenced_by_labels
```

Comment:
in this function, why does `K8sNamespace` need the member `prior_default_sidecar`?
can it use the same mechanism as for peers, with a member of `IstioSidecarYamlParser` that stores which namespaces were already configured by a parsed sidecar?

Reply:
Hi @tanyaveksler , since all code changes in this file (in current PR) are copied as is from your branch, I would like to know what you think about this.
also, I would like if you can review the documentation updates that I have committed  (last commit)

Reply:
committed changes with similar mechanism for the namespaces